### PR TITLE
VLCMovieViewController: Integrate MediaNavigationBar

### PR DESCRIPTION
**Pull request by robwayne**
### Checklist
- [x] I've run `bundle exec fastlane test` from the root directory to see all new and existing tests pass
- [x] I've followed the [vlc-ios code style](Docs/CodingStyle.md)
- [x] I've read the [Contribution Guidelines](https://github.com/videolan/vlc-ios#contribute)
- [x] I've updated the documentation if necessary.

### CODE REVIEW

File: Sources/MediaNavigationBar.swift [hunk 1]
```diff
@@ -77,59 +88,41 @@ protocol MediaNavigationBarDelegate {
...
-        ]
-        
-        NSLayoutConstraint.activate(constraints)
-    }
-    
     private func setupViews() {
-        addSubview(minimizePlaybackButton)
-        addSubview(mediaTitleTextLabel)
-        addSubview(chromeCastButton)
+        axis = .horizontal
```
Comment (Mikanbu) [inline Sources/MediaNavigationBar.swift]: `UILayoutConstraintAxisHorizontal` should already be the default on UIStackViews.

File: Sources/MediaNavigationBar.swift [hunk 2]
```diff
@@ -77,59 +88,41 @@ protocol MediaNavigationBarDelegate {
...
+        spacing = 20.0
+        alignment = .fill
```
Comment (Mikanbu) [inline Sources/MediaNavigationBar.swift]: Same remark as the axis.

File: Sources/MediaNavigationBar.swift [hunk 3]
```diff
@@ -77,59 +88,41 @@ protocol MediaNavigationBarDelegate {
...
+        assert(delegate != nil, "Delegate not set for MediaNavigationBar")
+        delegate?.mediaNavigationBarDidLongPressMinimize(self)
     }
     
     func toggleChromeCast() {
         assert(delegate != nil, "Delegate not set for MediaNavigationBar")
         delegate?.mediaNavigationBarDidToggleChromeCast(self)
-    }
-}
+    }}
```
Comment (Mikanbu) [inline Sources/MediaNavigationBar.swift]: 🤔

File: Resources/en.lproj/Localizable.strings [hunk 4]
```diff
@@ -378,3 +378,8 @@
 "MORE_OPTIONS_HINT" = "View more option controls";
 "REPEAT_MODE_HINT" = "Change repeat mode of current item";
 "VIDEO_ASPECT_RATIO_HINT" = "Change the aspect ratio of the current video";
+
+/* New strings */
+
+/* New strings */
+"MEDIA_NAV_TITLE" = "Media Title";
```
Comment (Mikanbu) [inline Resources/en.lproj/Localizable.strings]: I would remove `/* New strings */` or perhaps replace it by the used class/context?

File: Sources/VLCMovieViewController.m [hunk 5]
```diff
@@ -44,7 +45,7 @@
 #define DEFAULT_FOV 80.f
 #define MAX_FOV 150.f
 #define MIN_FOV 20.f
-#define NEW_UI 0
+#define NEW_UI 1
```
Comment (Mikanbu) [inline Sources/VLCMovieViewController.m]: I don't think this change should be committed.
Additionally, it seems that it doesn't compile with the "old" UI.

File: Resources/en.lproj/Localizable.strings (hunk 4 above)
Comment (robwayne) [reply Resources/en.lproj/Localizable.strings]: used class context?

Comment (Mikanbu) [reply Resources/en.lproj/Localizable.strings]: In this case it would be `MediaNavigationBar`

Comment (robwayne) [reply Resources/en.lproj/Localizable.strings]: oh haha u mean the comment not the localized string itself. I need to learn to read lol. Okay done!

**Comment by robwayne** [comment]
ping!

File: Sources/MediaNavigationBar.swift [hunk 6]
```diff
@@ -57,6 +66,7 @@ protocol MediaNavigationBarDelegate {
         var airPlayRoutePicker = AVRoutePickerView(frame: .zero)
         airPlayRoutePicker.activeTintColor = .orange
         airPlayRoutePicker.tintColor = .white
+        airPlayRoutePicker.widthAnchor.constraint(equalToConstant: buttonSize).isActive = true
```
Comment (carolanitz) [inline Sources/MediaNavigationBar.swift]: is this really necessary? don't the buttons resize themself based on the image?

File: Sources/VLCMovieViewController.m (hunk 5 above)
Comment (carolanitz) [reply Sources/VLCMovieViewController.m]: this should be 0 I agree

File: Sources/VLCMovieViewController.m [hunk 7]
```diff
@@ -54,7 +55,7 @@ typedef NS_ENUM(NSInteger, VLCPanType) {
   VLCPanTypeProjection
 };
 
-@interface VLCMovieViewController () <UIGestureRecognizerDelegate, VLCMultiSelectionViewDelegate, VLCEqualizerViewUIDelegate, VLCPlaybackControllerDelegate, VLCDeviceMotionDelegate, VLCRendererDiscovererManagerDelegate, PlaybackSpeedViewDelegate, VLCVideoOptionsControlBarDelegate>
+@interface VLCMovieViewController () <UIGestureRecognizerDelegate, VLCMultiSelectionViewDelegate, VLCEqualizerViewUIDelegate, VLCPlaybackControllerDelegate, VLCDeviceMotionDelegate, VLCRendererDiscovererManagerDelegate, PlaybackSpeedViewDelegate, VLCVideoOptionsControlBarDelegate, VLCMediaNavigationBarDelegate>
```
Comment (carolanitz) [inline Sources/VLCMovieViewController.m]: you know what? Looking at all the code that gets added I would add a Swift extension for MovieViewController that includes all the delegate methods for the new UI and other new UI code to have clearer distinguishment between new and old

File: Sources/VLCMovieViewController.m [hunk 8]
```diff
@@ -1383,7 +1435,11 @@ - (void)toggleUILock
     for (NSObject *item in items) {
         if ([item isKindOfClass:[UIControl class]]) {
             UIControl *control = (UIControl *)item;
-            control.enabled = !_interfaceIsLocked;
+            if (NEW_UI && item == _mediaNavigationBar.chromeCastButton) {
+                control.enabled = !_interfaceIsLocked && _vpc.renderer != nil;
```
Comment (carolanitz) [inline Sources/VLCMovieViewController.m]: that looks wrong. The UILock should be not effected by having a renderer

File: Sources/MediaNavigationBar.swift [hunk 9]
```diff
@@ -48,7 +55,9 @@ protocol MediaNavigationBarDelegate {
         chromeButton.addTarget(self, action: #selector(toggleChromeCast), for: .touchUpInside)
         chromeButton.setImage(UIImage(named: "renderer"), for: .normal)
         chromeButton.tintColor = .white
+        chromeButton.widthAnchor.constraint(equalToConstant: buttonSize).isActive = true
         chromeButton.translatesAutoresizingMaskIntoConstraints = false
+        chromeButton.isEnabled = false // disable until a chromecast device is found
```
Comment (carolanitz) [inline Sources/MediaNavigationBar.swift]: that button should not be visible I assume when there is no chromecast device

File: Sources/MediaNavigationBar.swift [hunk 10]
```diff
@@ -13,32 +13,39 @@ import AVKit
...
     
-    private var mediaTitleTextLabel: UILabel = {
+    lazy var mediaTitleTextLabel: UILabel = {
         var label = UILabel()
         label.numberOfLines = 1
         label.lineBreakMode = .byTruncatingTail
         label.textColor = .white
         label.font = UIFont(name: "SFProDisplay-Medium", size: 17)
-        label.text = NSLocalizedString("TITLE", comment: "Video Title")
+        label.text = NSLocalizedString("MEDIA_NAV_TITLE", comment: "")
```
Comment (carolanitz) [inline Sources/MediaNavigationBar.swift]: I would say this needs to be empty unless we have a media title?

File: Sources/MediaNavigationBar.swift (hunk 9 above)
Comment (robwayne) [reply Sources/MediaNavigationBar.swift]: hmm would it not be just as okay to have the button but not have it selectable if a renderer isnt present?

File: Sources/VLCMovieViewController.m (hunk 8 above)
Comment (robwayne) [reply Sources/VLCMovieViewController.m]: This is actually setting the chromecast button to be enabled only if there is a renderer device and the interface lock is not locked. aka if the interface is unlocked but no renderer it remains disabled

File: Sources/MediaNavigationBar.swift (hunk 6 above)
Comment (robwayne) [reply Sources/MediaNavigationBar.swift]: Actually despite trying setContentHuggingPriority on the buttons, the minimize and chromecast buttons are unwilling to stay up without the widths set. "Width and horizontal position are ambiguous"

File: Sources/VLCMovieViewController.m (hunk 7 above)
Comment (robwayne) [reply Sources/VLCMovieViewController.m]: Okay I'll follow this PR up with another PR where I do this, seeing as I have other PRs out with additional code that would need to be added as well.

**Comment by Mikanbu** [comment]
Merged with 92c0b74b8c6e7817faa12a95476242cad066a2df.

**Pull request closed without merging**